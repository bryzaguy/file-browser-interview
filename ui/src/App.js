import * as React from 'react';
import PropTypes from 'prop-types';
import FilterableList from './components/FilterableList';
import { Box, useMediaQuery } from '@mui/material';
import { createTheme, ThemeProvider } from '@mui/material/styles';

export default function App({ api }) {
  const [prefix, setPrefix] = React.useState(null);
  const [items, setItems] = React.useState([]);
  const [errorMessage, setErrorMessage] = React.useState(null);

  // Calls the api initially and when prefix changes.
  React.useEffect(() => {
    setItems([]);
    setErrorMessage('');
    api({ prefix }).then(
      result => { setItems(result) },
      () => setErrorMessage("Problem loading data.")
    );
  }, [prefix, api]);

  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
  const mdTheme = createTheme({
    palette: {
      mode: prefersDarkMode ? 'dark' : 'light'
    }
  });

  return (
    <ThemeProvider theme={mdTheme}>
      <Box
        width='100%'
        minHeight='100vh'
        sx={{
          backgroundColor: (theme) =>
            theme.palette.mode === 'light'
              ? theme.palette.grey[100]
              : theme.palette.grey[900]
        }}>
        <Box maxWidth={800} p='1rem'
          sx={{ 'margin': 'auto' }}
          noValidate
          autoComplete="off"
        >
          <FilterableList
            items={items}
            onFilter={e => setPrefix(e.target.value)}
            errorMessage={errorMessage}
          />
        </Box>
      </Box>
    </ThemeProvider>
  );
}

App.propTypes = {
  api: PropTypes.func.isRequired
}

