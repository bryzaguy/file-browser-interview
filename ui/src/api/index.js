const cache = {};

export default function api(args = {}) {
  // Filter args with no value.
  const params = Object.keys(args).reduce((result, key) => {
    if (args[key]) {
      result[key] = args[key];
    }
    return result
  }, {});

  const url = 'http://localhost:8081/search?' + new URLSearchParams(params);
  const clearFailedPromise = () => { cache[url] = null };

  if (!cache[url]) {
    cache[url] = fetch(url)
      .then(res => res.json(), clearFailedPromise)
      .then(res => res.results);
  }

  return cache[url];
}