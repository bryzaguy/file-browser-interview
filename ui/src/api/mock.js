import data from './mockData.json';

export default function TestApi({ prefix } = {}) {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(data.results.filter(name => !prefix || name.startsWith(prefix)));
    });
  });
}