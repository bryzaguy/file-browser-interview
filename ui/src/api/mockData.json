{
  "results": [
    ".yarn-integrity",
    "accepts/HISTORY.md",
    "accepts/LICENSE",
    "accepts/README.md",
    "accepts/index.js",
    "accepts/package.json",
    "array-flatten/LICENSE",
    "array-flatten/README.md",
    "array-flatten/array-flatten.js",
    "array-flatten/package.json"
  ]
}