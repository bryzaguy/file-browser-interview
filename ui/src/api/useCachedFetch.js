import * as React from 'react';
const cache = {};

function cachedFetch(url, onSuccess) {
  const onFailure = (e) => { cache[url] = null; throw e };
  cache[url] = cache[url] || fetch(url).then(onSuccess, onFailure);
  return cache[url];
}

export default function useCachedFetch(defaultData) {
  const [error, setError] = React.useState('');
  const [loading, setLoading] = React.useState(false);
  const [initialData] = React.useState(defaultData);
  const [data, setData] = React.useState(initialData);
  const [promise, setPromise] = React.useState(null);

  const fetch = React.useCallback((url, onSuccess) => {
    setPromise(cachedFetch(url, onSuccess));
  }, []);

  React.useEffect(() => {
    if (!promise) return;

    setData(initialData);
    setLoading(true);
    setError('');

    var isLastestPromise = true;
    promise.then(result => {
      if (isLastestPromise) {
        setData(result);
        setLoading(false);
      }
    }, () => {
      if (isLastestPromise) {
        setError("There was a problem fetching data.");
        setLoading(false);
      }
    });

    return () => { isLastestPromise = false; };
  }, [promise, initialData]);

  return { fetch, loading, error, data };
}
