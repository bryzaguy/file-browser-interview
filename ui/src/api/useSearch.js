import * as React from 'react';
import useCachedFetch from './useCachedFetch';

function removeEmptyKeys(args = {}) {
  return Object.keys(args).reduce((result, key) => {
    if (args[key]) {
      result[key] = args[key];
    }
    return result;
  }, {});
}

const SEARCH_URL = process.env.REACT_APP_SEARCH_URL;

function getSearchUrl(args) {
  const params = removeEmptyKeys(args);
  return `${SEARCH_URL}?${new URLSearchParams(params)}`;
}

export default function useSearch() {
  const { fetch, loading, error, data } = useCachedFetch([]);

  const get = React.useCallback(args => {
    const url = getSearchUrl(args);
    fetch(url, res => res.json().then(json => json.results));
  }, [fetch]);

  return { get, loading, error, data };
}
