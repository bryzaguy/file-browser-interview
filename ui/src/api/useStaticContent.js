import * as React from 'react';
import useCachedFetch from './useCachedFetch';

const STATIC_URL = process.env.REACT_APP_STATIC_URL;

const getStaticUrl = name => `${STATIC_URL}${name}`;

export default function useStaticContent() {
  const { fetch, loading, error, data } = useCachedFetch(null);

  const get = React.useCallback(name => {
    const url = getStaticUrl(name);
    fetch(url, res => res.text());
  }, [fetch]);

  return { get, loading, error, data };
}
