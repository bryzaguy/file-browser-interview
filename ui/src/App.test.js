import { render } from '@testing-library/react';
import App from './App';
import api from './api/mock';

const originalError = console.error;

beforeAll(() => {
  console.error = (...args) => {
    if (/Warning.*(Use createRoot instead|not wrapped in act)/.test(args[0])) {
      return;
    }
    originalError.call(console, ...args);
  }
});

afterAll(() => {
  console.error = originalError;
});

test('renders app', (done) => {
  const app = render(<App api={api} />);
  setTimeout(() => {
    expect(app).toMatchSnapshot();
    done();
  })
});
