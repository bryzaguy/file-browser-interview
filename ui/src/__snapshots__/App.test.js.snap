// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1ie02b"
      >
        <div
          autocomplete="off"
          class="MuiBox-root css-7qauub"
          novalidate=""
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
          >
            <div
              class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
            >
              <table
                class="MuiTable-root css-rqglhn-MuiTable-root"
              >
                <thead
                  class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                      colspan="2"
                      scope="col"
                    >
                      <div
                        class="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1s97tdc-MuiInputBase-root-MuiInput-root"
                      >
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="FilterAltOutlinedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 6h10l-5.01 6.3L7 6zm-2.75-.39C6.27 8.2 10 13 10 13v6c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-6s3.72-4.8 5.74-7.39c.51-.66.04-1.61-.79-1.61H5.04c-.83 0-1.3.95-.79 1.61z"
                            />
                          </svg>
                        </div>
                        <input
                          class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                          type="text"
                          value=""
                        />
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                          />
                        </div>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      .yarn-integrity
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      accepts/HISTORY.md
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      accepts/LICENSE
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      accepts/README.md
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      accepts/index.js
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      accepts/package.json
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      array-flatten/LICENSE
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      array-flatten/README.md
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      array-flatten/array-flatten.js
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="InsertDriveFileOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                        />
                      </svg>
                        
                      array-flatten/package.json
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1ie02b"
    >
      <div
        autocomplete="off"
        class="MuiBox-root css-7qauub"
        novalidate=""
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
        >
          <div
            class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    colspan="2"
                    scope="col"
                  >
                    <div
                      class="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1s97tdc-MuiInputBase-root-MuiInput-root"
                    >
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="FilterAltOutlinedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 6h10l-5.01 6.3L7 6zm-2.75-.39C6.27 8.2 10 13 10 13v6c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-6s3.72-4.8 5.74-7.39c.51-.66.04-1.61-.79-1.61H5.04c-.83 0-1.3.95-.79 1.61z"
                          />
                        </svg>
                      </div>
                      <input
                        class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                        />
                      </div>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    .yarn-integrity
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    accepts/HISTORY.md
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    accepts/LICENSE
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    accepts/README.md
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    accepts/index.js
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    accepts/package.json
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    array-flatten/LICENSE
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    array-flatten/README.md
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    array-flatten/array-flatten.js
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root css-1oqrj23-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-kw1evd-MuiTableCell-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="InsertDriveFileOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                      />
                    </svg>
                      
                    array-flatten/package.json
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
